app/uk/gov/hmrc/ups/connectors/PreferencesConnector.scala

```diff
@@ -43,48 +49,35 @@ trait PreferencesConnector {
       }
   }
 
-  implicit val statusReads = new HttpReads[Int] {
+  implicit val statusReads: HttpReads[Int] = new HttpReads[Int] {
     def read(method: String, url: String, response: HttpResponse): Int = response.status
   }
 
-  def pullWorkItem(implicit hc: HeaderCarrier): Future[Option[PulledItem]] =
-    http.POST[WorkItemRequest, Option[PulledItem]](
+  def pullWorkItem(implicit hc: HeaderCarrier): Future[Option[PulledItem]] = {
+    httpClient.POST[WorkItemRequest, Option[PulledItem]](
       s"$serviceUrl/preferences/updated-print-suppression/pull-work-item",
       workItemRequest
     ).recover { case ex =>
       Logger.error(s"Call to $serviceUrl/preferences/updated-print-suppression/pull-work-item failed unexpectedly", ex)
       None
     }
+  }
 
-  def changeStatus(callbackUrl: String, status: ProcessingStatus)(implicit hc: HeaderCarrier) =
-    http.POST[JsValue, Int](s"$serviceUrl$callbackUrl", Json.obj("status" -> status.name))
-
-  def retryFailedUpdatesAfter: Duration
+  def changeStatus(callbackUrl: String, status: ProcessingStatus)(implicit hc: HeaderCarrier): Future[Int] =
+    httpClient.POST[JsValue, Int](s"$serviceUrl$callbackUrl", Json.obj("status" -> status.name))
 
-  def dateTimeFor(duration: Duration): DateTime = DateTimeUtils.now.minus(duration)
+  def dateTimeFor(duration: Duration): DateTime = DateTimeUtils.now.minus(duration.toMillis)
```

Comment:
Do we need to specify implicits for joda reads and writes? Not sure if this is relevant here

Reply:
Updated.